.pre-commit-config.yaml

```diff
@@ -1,24 +1,112 @@
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v5.0.0
+    rev: v6.0.0
     hooks:
       - id: check-added-large-files
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: check-case-conflict
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: check-merge-conflict
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: check-symlinks
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: check-toml
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: check-yaml
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: end-of-file-fixer
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: mixed-line-ending
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: trailing-whitespace
-        exclude: \.md$
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: no-commit-to-branch
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
+
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.11.12
+    rev: v0.14.1
     hooks:
       - id: ruff-check
         args: [--fix]
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
       - id: ruff-format
+        exclude: |
+          (?x)^(
+            .*\.md$|
+            .*\.svg$
+          )
+
+  - repo: https://github.com/trufflesecurity/trufflehog
```

Comment:
Could you rebase on main, I've also redone this implementation and forced only checking for files that have been commited, so this should resolve a huge part of it too. Perhaps we can also raise a certain error if this fails and redirect users to the install page?